docs(app_check): add additional steps to ensure App Check works on specific devices (#17110)
firebase/flutterfire: docs/app-check/default-providers.md

App Check debug provider instead of a real attestation provider.

See [Use App Check with the debug provider in Flutter apps](/docs/app-check/flutter/debug-provider).

> Note: For certain Android devices, you need to enable "Meets basic device integrity" in the 
> Play console.

Reviewer: ```suggestion
Note: For certain Android devices, you may need to enable "Meets basic device integrity" in the 
Google Play console.
```

App Check debug provider instead of a real attestation provider.

See [Use App Check with the debug provider in Flutter apps](/docs/app-check/flutter/debug-provider).

> ### Additional Steps to Enable App Check on Specific Devices (e.g. Xiaomi)

Reviewer: ```suggestion

```

> For certain devices like Xiaomi, you need to enable "Meets basic device integrity" in the 

Reviewer: ```suggestion
Note: For certain Android devices, you need to enable "Meets basic device integrity" in the 
```

> Play Integrity API settings, as mentioned in [this issue comment](https://github.com/firebase/flutterfire/issues/17106#issuecomment-2663422877), 

Reviewer: ```suggestion
Play console
```
